specification/customer-insights/resource-manager/Microsoft.CustomerInsights/2017-04-26/customer-insights.json

```diff
@@ -3149,6 +3149,378 @@
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "name": "predictionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Prediction."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Successfully get the training results.",
+            "schema": {
+              "$ref": "#/definitions/PredictionTrainingResults"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/predictions/{predictionName}/getModelStatus": {
+      "post": {
+        "tags": [
+          "Predictions"
+        ],
+        "operationId": "Predictions_GetModelStatus",
+        "x-ms-examples": {
+          "Predictions_GetModelStatus": {
+            "$ref": "./examples/PredictionsGetModelStatus.json"
+          }
+        },
+        "description": "Gets model status of the prediction.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "name": "predictionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Prediction."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Successfully get the model status.",
+            "schema": {
+              "$ref": "#/definitions/PredictionModelStatus"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/predictions/{predictionName}/modelStatus": {
+      "post": {
+        "tags": [
+          "Predictions"
+        ],
+        "operationId": "Predictions_ModelStatus",
```

Comment:
You are right. Not all of these status can be set to. Do you think I should create another enum for this? Or I can leave this since our service will return validation error if invalid status is set?